espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt

```diff
@@ -278,9 +350,37 @@ class SourceDocument constructor(
             toStream().use { xmlStream ->
                 val list = expr.evaluate(InputSource(xmlStream), XPathConstants.NODESET) as NodeList
                 (0 until list.length).map { index ->
-                    viewMap.get(Integer.parseInt((list.item(index) as Element).getAttribute(VIEW_INDEX)))
+                    viewMap.get(
+                        Integer.parseInt(
+                            (list.item(index) as Element).getAttribute(
+                                VIEW_INDEX
+                            )
+                        )
+                    )
                 }
             }
         }, { performCleanup() })
     }
+
+    fun hasXpath(xpathSelector: String): SemanticsMatcher {
```

Comment:
private?

Reply:
It is being used in another file

Reply:
I'm not quite sure this method logically belongs to SourceDocument entity. Maybe it makes sense to extract it to a separate module?

Reply:
I can move it to `ComopseNodeFinder.kt`. This will demand some methods of `SourceDocument ` to be made public/internal

Reply:
I kept it there because there was counterpart method `SourceDocument#findViewsByXPath` for espresso